sources/web/datalab/polymer/modules/Utils.ts

```diff
@@ -41,32 +37,40 @@ class Utils {
 
   /**
    * Opens a dialog with the specified options. It uses the Datalab custom element
-   * <input-dialog>, attaches a new instance to the current document, opens it
-   * and returns a promise that resolves when the dialog is closed.
-   * @param dialogOptions options for configuring the dialog
+   * according to the specified dialog type, attaches a new instance to the current
+   * document, opens it, and returns a promise that resolves when the dialog is closed.
+   * @param type specifies which type of dialog to use
+   * @param dialogOptions specifies different options for opening the dialog
    */
-  static getUserInputAsync(dialogOptions: DialogOptions) {
-    const createModal = <InputDialogElement>document.createElement('input-dialog');
-    document.body.appendChild(createModal);
+  static showDialog(type: DialogType, dialogOptions: DialogOptions) {
+    let dialogElement = '';
+    if (type === DialogType.input) {
```

Comment:
When using an OO system, case statements and long if/else chains are usually a sign to look for an OO solution that allows new choices or types to be added just by adding another subclass. In this case, since your if/else chain is only two items long, and I think this may be the only place where you are checking the value of the DialogType enum, it's not a problem.